### PR TITLE
 Partial execute

### PR DESCRIPTION
**Pull request by TymofiiSavinov**

### CODE REVIEW

**Comment by markblaize** [comment]
Task linked: [CU-861mf22bv Partial execute](https://app.clickup.com/t/861mf22bv)

File: contracts/leverage_trading/src/cancel_limit_order.rs [hunk 1]
```diff
@@ -82,15 +79,44 @@ impl Contract {
...
+                    panic!("Some problem with return amount from Dex.")
+                }
+            }
+            _ => panic!("DEX not found liquidity amounts."),
+        };
+
+        self.increase_balance(
+            &signer_account_id(),
+            &order.sell_token,
+            return_liquidity_amounts.get(0).unwrap().0,
```
Comment (arsenijkovalov) [inline contracts/leverage_trading/src/cancel_limit_order.rs:96]: Amount should be converted to protocol token decimals

File: contracts/leverage_trading/src/cancel_limit_order.rs [hunk 2]
```diff
@@ -82,15 +79,44 @@ impl Contract {
...
+        );
+        self.increase_balance(
+            &signer_account_id(),
+            &order.buy_token,
+            return_liquidity_amounts.get(1).unwrap().0,
```
Comment (arsenijkovalov) [inline contracts/leverage_trading/src/cancel_limit_order.rs:101]: Amount should be converted to protocol token decimals

File: contracts/leverage_trading/src/cancel_take_profit_order.rs [hunk 3]
```diff
@@ -143,17 +115,43 @@ impl Contract {
...
+                    panic!("Some problem with return amount from Dex.")
+                }
+            }
+            _ => panic!("DEX not found liquidity amounts."),
+        };
+
+        self.increase_balance(
+            &signer_account_id(),
+            &parent_order.sell_token,
+            return_liquidity_amounts.get(0).unwrap().0,
```
Comment (arsenijkovalov) [inline contracts/leverage_trading/src/cancel_take_profit_order.rs:147]: Amount should be converted to protocol token decimals

File: contracts/leverage_trading/src/cancel_take_profit_order.rs [hunk 4]
```diff
@@ -143,17 +115,43 @@ impl Contract {
...
+        );
+        self.increase_balance(
+            &signer_account_id(),
+            &parent_order.buy_token,
+            return_liquidity_amounts.get(1).unwrap().0,
```
Comment (arsenijkovalov) [inline contracts/leverage_trading/src/cancel_take_profit_order.rs:152]: Amount should be converted to protocol token decimals

**Review by arsenijkovalov — APPROVED** [review]

**Pull request merged**